mlir/lib/Gradient/Transforms/GradMethods/ClassicalJacobian.cpp

```diff
@@ -160,10 +164,12 @@ func::FuncOp genArgMapFunction(PatternRewriter &rewriter, Location loc, func::Fu
 
                 rewriter.replaceOp(op, gate.getQubitOperands());
             }
-            // The function does not return anything (the parameter vector is passed in as an
-            // argument.)
             else if (auto returnOp = dyn_cast<func::ReturnOp>(op)) {
-                returnOp.getOperandsMutable().clear();
+                PatternRewriter::InsertionGuard insertionGuard(rewriter);
+                rewriter.setInsertionPoint(returnOp);
+                Value paramsVector =
+                    rewriter.create<bufferization::ToTensorOp>(loc, paramsVectorType, paramsBuffer);
+                returnOp.getOperandsMutable().assign(paramsVector);
```

Comment:
Is it a reset to what it was previously? Is it necessary for argmap to not be dp style?

Reply:
Yes it's just resetting to what it was previously. The motivation here was to make the argmap either have just tensors or just memrefs, not a combination. This makes it easier to check that the BackpropOp has the right argument and return types.

Reply:
Thanks !